src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -1609,6 +1646,88 @@ RebalanceTableShards(RebalanceOptions *options, Oid shardReplicationModeOid)
 }
 
 
+/*
+ * RebalanceTableShardsBackground rebalances the shards for the relations
+ * inside the relationIdList across the different workers. It does so using our
+ * background job+task infrastructure.
+ */
+static void
+RebalanceTableShardsBackground(RebalanceOptions *options, Oid shardReplicationModeOid)
+{
+	char transferMode = LookupShardTransferMode(shardReplicationModeOid);
+	EnsureReferenceTablesExistOnAllNodesExtended(transferMode);
+
+	if (list_length(options->relationIdList) == 0)
+	{
+		return;
+	}
+
+	Oid relationId = InvalidOid;
+	char *operationName = "rebalance";
+	if (options->drainOnly)
+	{
+		operationName = "move";
+	}
+
+	foreach_oid(relationId, options->relationIdList)
+	{
+		AcquireRebalanceColocationLock(relationId, operationName);
```

Comment:
Would be good to add a comment that this is to fail in case a concurrent rebalance_table_shards is running